ads/google/a4a/utils.js

```diff
@@ -361,6 +354,25 @@ export function additionalDimensions(win, viewportSize) {
 };
 
 /**
+ * Returns a map containing all ad containers that enclose the given ad
+ * element. The params of the map are restricted to the ValidAdContainerTypes,
+ * and the values are all true.
+ * @param {!Element} adElement
+ * @return {!Object<string, boolean>}
+ */
+export function getAllParentAdContainers(adElement) {
```

Comment:
This was a decision Keith made when I first added the 'act' URL parameter. In short, we'd rather check a non-exhaustive set of ancestors than get stuck looping for a long time.